### PR TITLE
feat(CI):enable markdownlint and typos in docs.yml

### PR DESCRIPTION
**Pull request by ywh555hhh**
Improve CI flow, for docs.yml, enable markdownlint and typos

### Reason for this PR

to fix issue #499 

### What changes are included in this PR?

enable markdownlint and typos in docs.yml

### Are these changes tested?

no local test

### Are there any user-facing changes?

no

### CODE REVIEW

**Comment by SemyonSinchenko** [comment]
May we also make a script for do that before commit? For example, in folder `./dev` something like `fix_spelling.sh`?

**Comment by ywh555hhh** [comment]
I'll try to fix it automatically with markdownlint

**Comment by ywh555hhh** [comment]
When using Markdownlint for document formatting checks, I encountered some issues that could not be automatically resolved, mainly involving a large number of MD013 (line length is too long) and MD033 (inline HTML) issues, as well as one MD025 (multiple top-level headings) issue.

For example, I encountered MD013 and MD033 issues on line 110 of the `specification/format.md` file, and an MD025 issue in the `libraries/cpp/getting-started.md` file.

To resolve these issues, I have configured Markdownlint to ignore these three types of issues in the `.markdownlint.yaml` configuration file. The specific configuration is as follows:

```yaml
# Ignore MD013, as the document requires longer lines to maintain the integrity of code examples
MD013: false

# Ignore MD033, as inline HTML is necessary in some cases, such as specific formatting requirements
MD033: false

# Ignore MD025, as the document structure requires multiple top-level headings to reflect different sections or parts
MD025: false
```

I believe this is a reasonable solution because these issues are mainly caused by the mismatch between our document format and the default rules of Markdownlint. By customizing our Markdownlint configuration, we can ensure that it better adapts to our document structure and formatting needs.

File: docs/.markdownlint.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# 忽略 MD013，因为文档需要较长的行以保持代码示例的完整性
```
Comment (acezen) [inline docs/.markdownlint.yaml:1]: It's better to use english as document language

**Review by acezen — COMMENTED** [review]
Hi, @ywh555hhh, it seems that the markdown format work, but there still some format error in CI, could you help to fix them? then the PR can be merged.

**Comment by ywh555hhh** [comment]
> Hi, @ywh555hhh, it seems that the markdown format work, but there still some format error in CI, could you help to fix them? then the PR can be merged.

I expect to have time to deal with it next week

**Comment by yecol** [comment]
Cool ~ Enhanced CI to improve the quality of our documentations! LGTM. :)

**Review by acezen — APPROVED** [review]
LGTM, thanks @ywh555hhh

**Comment by acezen** [comment]
Hi, @ywh555hhh, I have a question, why this change affect to the testing submodule?

**Review by acezen — CHANGES_REQUESTED** [review]
the testing submodule should not be modified

**Comment by ywh555hhh** [comment]
> Hi, @ywh555hhh, I have a question, why this change affect to the testing submodule?

Seems I ran some tool locally incorrectly

**Comment by ywh555hhh** [comment]
> the testing submodule should not be modified

testing submodule is no longer affected

**Review by acezen — APPROVED** [review]
LGTM, thanks very much for the contribution!

**Pull request merged**